nrrd/reader.py

```diff
@@ -420,25 +425,37 @@ def read_data(header, fh=None, filename=None):
         raise NRRDError('Size of the data does not equal the product of all the dimensions: {0}-{1}={2}'
                         .format(total_data_points, data.size, total_data_points - data.size))
 
-    # Eliminate need to reverse order of dimensions. NRRD's data layout is the same as what Numpy calls 'Fortran'
-    # order, where the first index is the one that changes fastest and last index changes slowest
-    data = np.reshape(data, tuple(header['sizes']), order='F')
+    # In the NRRD header, the fields are specified in Fortran order, i.e, the first index is the one that changes
+    # fastest and last index changes slowest. This needs to be taken into consideration since numpy uses C-order
+    # indexing.
+    
+    # The array shape from NRRD (x,y,z) needs to be reversed as numpy expects (z,y,x).
+    data = np.reshape(data, tuple(header['sizes'][::-1]))
+
+    # Transpose data to enable Fortran indexing if requested.
+    if index_order == 'F':
+        data = data.T
 
     return data
 
 
-def read(filename, custom_field_map=None):
+def read(filename, custom_field_map=None, index_order='F'):
     """Read a NRRD file and return the header and data
 
     See :ref:`user-guide:Reading NRRD files` for more information on reading NRRD files.
 
+    .. note::
+            Users should be aware that the `index_order` argument needs to be consistent between `nrrd.read` and `nrrd.write`. I.e., reading an array with `index_order='F'` will result in a transposed version of the original data and hence the writer needs to be aware of this.
+
     Parameters
     ----------
     filename : :class:`str`
         Filename of the NRRD file
     custom_field_map : :class:`dict` (:class:`str`, :class:`str`), optional
         Dictionary used for parsing custom field types where the key is the custom field name and the value is a
         string identifying datatype for the custom field.
+    index_order : :class: `str`, optional
+        Specifies the index ordering of the outputted array, either 'F' (Fortran order) or 'C' (C order). Using 'F' (default) the image will be transposed, providing an index ordering in this manneri; `data[x][y][z]`. However, Numpy generally uses 'C' order, (e.g. `data[z][y][x]`) and 'C' order is therefore adviced to keep consistent with other libraries. 
```

Comment:
I think this could cause some confusion in the same manner as you mentioned in a previous comment. "Specifies the index order used for reading" implies that you are reading in a certain order from the file, which is not the case. The order is simply the index order (whether to transpose the data or not before returning).

Reply:
You mentioned that "slowest-varying to fastest-varying" is vague without a description of memory layout which I agree with. Whatever language we come up with to describe C-order & Fortran-order, it should be here.

Fair enough, how about the below wording? It's basically changing "outputted" to "resulting data"
"Specifies the index order of the resulting data array."